code/game/machinery/dish_drive.dm

```diff
@@ -0,0 +1,139 @@
+/obj/machinery/dish_drive
+	name = "dish drive"
+	desc = "A culinary marvel that uses matter-to-energy conversion to store dishes and shards. Convenient! \
+	Additional features include a vacuum function to suck in nearby dishes, and an automatic transfer beam that empties its contents into nearby disposal bins every now and then. \
+	Or you can just drop your plates on the floor, like civilized folk."
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "synthesizer"
+	idle_power_usage = 8 //5 with default parts
+	active_power_usage = 13 //10 with default parts
+	anchored = FALSE
+	density = FALSE
+	pass_flags = PASSTABLE
+	var/static/list/collectable_items = list(/obj/item/trash/waffles,
+		/obj/item/trash/plate,
+		/obj/item/trash/tray,
+		/obj/item/reagent_containers/food/drinks/drinkingglass,
+		/obj/item/kitchen/utensil/fork,
+		/obj/item/shard,
+		/obj/item/broken_bottle)
+	var/static/list/disposable_items = list(/obj/item/trash/waffles,
+		/obj/item/trash/plate,
+		/obj/item/trash/tray,
+		/obj/item/shard,
+		/obj/item/broken_bottle)
+	var/time_since_dishes = 0
+	var/suction_enabled = TRUE
+	var/transmit_enabled = TRUE
+	var/list/dish_drive_contents
+
+/obj/machinery/dish_drive/Initialize(mapload)
+	. = ..()
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/dish_drive(null)
+	component_parts += new /obj/item/stock_parts/manipulator(null)
+	component_parts += new /obj/item/stock_parts/matter_bin(null)
+	component_parts += new /obj/item/stack/sheet/glass(null)
+	RefreshParts()
+
+/obj/machinery/dish_drive/examine(mob/user)
+	. = ..()
+	if(user.Adjacent(src))
+		. += "<span class='notice'>Alt-click it to beam its waste storage contents to any nearby disposal bins.</span>"
+
+/obj/machinery/dish_drive/attack_hand(mob/living/user)
+	if(!LAZYLEN(dish_drive_contents))
+		to_chat(user, "<span class='warning'>There's nothing in [src]!</span>")
+		return
+	var/obj/item/I = LAZYACCESS(dish_drive_contents, LAZYLEN(dish_drive_contents)) //the most recently-added item
+	LAZYREMOVE(dish_drive_contents, I)
+	user.put_in_hands(I)
+	to_chat(user, "<span class='notice'>You take out [I] from [src].</span>")
+	playsound(src, 'sound/items/pshoom.ogg', 15, TRUE)
+	flick("synthesizer_beam", src)
+
+/obj/machinery/dish_drive/screwdriver_act(mob/user, obj/item/I)
+	if(default_deconstruction_screwdriver(user, "[initial(icon_state)]-o", initial(icon_state), I))
+		return TRUE
+
+/obj/machinery/dish_drive/wrench_act(mob/user, obj/item/I)
+	if(default_unfasten_wrench(user, I))
+		return TRUE
+
+/obj/machinery/dish_drive/crowbar_act(mob/user, obj/item/I)
+	if(default_deconstruction_crowbar(user, I))
+		return TRUE
+	..()
```

Comment:
Actually you can just remove this entire override since the parent does the same thing.